### DISCUSSION TITLE
Allow more flexibility in reception status state changes for return items

### DISCUSSION THREAD

**Discussion by RyanQuey** [solidusio/solidus#4588] [New Features or Ideas]
**Is your feature request related to a problem? Please describe.**
 It seems like there should be more flexibility built into the state machine for reception statuses of return items. For example, I would tend to think that one could transition from ["in transit" to "lost in transit" or "wrong item shipped"](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/return_item.rb#L90)

**Describe the solution you'd like**

Is there any reason why it is set up this way? I fully acknowledge that there is a good chance I said we don't understand how this state machine is supposed to function, in which case any advice is much appreciated :-)

**Comment by kennyadsl** [comment]
@RyanQuey thanks for your feedback. I think you are right, that state machine is not flexible at all. The issue with state machines in core models is that they are very hard to customize in stores implementation. Probably when this setup was done, it has been preferred to keep it simple leaving more flexibility to developers to add transitions and states, which is easier than removing them. 

Also, some work has been done to remove them at all (eg. https://github.com/solidusio/solidus/pull/3040) but I'm personally not sure it's the right path. 

If you have any feedback about how we can improve this part, it would be appreciated!
